crypto/pkcs12/pkcs12-example.c

```diff
@@ -87,42 +86,50 @@ int main(int argc, char** argv)
     ret = wc_PKCS12_parse(pkcs12, "wolfSSL test", &keyDer, &keySz,
             &certDer, &certSz, &list);
     printf("return value of parsing pkcs12 = %d %s\n", ret, (ret == 0)? "SUCCESS": "FAIL");
-    if (ret != 0 || keyDer == NULL || certDer == NULL) {
+    if (ret != 0) {
         printf("\t error parsing pkcs12\n");
         wc_PKCS12_free(pkcs12);
         return -1;
     }
 
     /* print out key and cert found */
-    printf("HEX of Private Key Read (DER format) :\n");
-    for (i = 0; i < keySz; i++) {
-        if (i != 0 && !(i%16)) printf("\n");
-        printf("%02X", keyDer[i]);
-    }
-    printf("\n");
-
-    printf("\nHEX of Certificate Read (DER format) :\n");
-    for (i = 0; i < certSz; i++) {
-        if (i != 0 && !(i%16)) printf("\n");
-        printf("%02X", certDer[i]);
-    }
-    printf("\n");
-
     if (keyDer != NULL) {
+        printf("HEX of Private Key Read (DER format) :\n");
+        for (i = 0; i < keySz; i++) {
+            if (i != 0 && !(i%16)) printf("\n");
+            printf("%02X", keyDer[i]);
+        }
+        printf("\n");
         XFREE(keyDer, NULL, DYNAMIC_TYPE_PKCS);
     }
 
     if (certDer != NULL) {
+        printf("\nHEX of Certificate Read (DER format) :\n");
+        for (i = 0; i < certSz; i++) {
+            if (i != 0 && !(i%16)) printf("\n");
+            printf("%02X", certDer[i]);
+        }
+        printf("\n");
         XFREE(certDer, NULL, DYNAMIC_TYPE_PKCS);
     }
 
     /* itterate through list if was not passed as null and free each node */
     if (list != NULL) {
         WC_DerCertList* current;
+        int certIdx = 0;
+
+        printf("\nHEX of Certificate LIST (DER format) :\n");
         current = list;
         while (current != NULL) {
-            WC_DerCertList* next = current->next;
+            WC_DerCertList* next;
+
+            next  = current->next;
             if (current->buffer != NULL) {
+                 printf("[CERT %d] :", certIdx++);
+                 for (i = 0; i < current->bufferSz; i++)
+                     printf("%02X", current->buffer[i]);
```

Comment:
The certificate HEX data is printed differently than the data for previous certificates.  Specifically, the previous HEX data was displayed in 16-byte rows.  The way the certificate list data is currently displayed makes the output look inconsistent, and there might also be some utility to having 16-byte rows, as it's easier to visually parse to a specific location (?).  

I'd suggest modifying the certificate list HEX data print logic to match that used for other HEX data.   I realize that the `[CERT %d] :` text printed prior to each certificate will push the first 16-byte row out of alignment with with subsequent rows... I'd suggest adding a newline to the end of the "CERT" text so the first row of HEX data for each certificate starts at column zero.